add scry docs

tinnus-napbus opened the pull request:
- adds a scry section in /arvo/concepts

The doc largely has a practical focus, filling in the gaps of the dotket documentation and clarifying some confusing aspects of scries. Links out to a couple of relevant sections as well. 

resolves #788

tinnus-napbus committed:
add scry docs

adds a scry section in /arvo/concepts

tinnus-napbus committed:
scry: fix typo

tinnus-napbus committed:
scries: change diagram url to media.urbit.org

Fang- commented on content/docs/arvo/concepts/scry.md:
```diff
@@ -0,0 +1,77 @@
+
+For details of its usage, see the [dotket](/docs/hoon/reference/rune/dot#-dotket) section of the Nock rune documentation. In addition to the dotket documentation, below is a colour-coded diagram breaking down the structure of a dotket scry expression with some additional notes.
+
+![Scry diagram](https://media.urbit.org/docs/arvo/scry-diagram.svg)
```
> Clay submode and/or the kind of data

Might be more accurate to say it signifies the kind of _request_? Within results, for the `x` care, there's still a lot of variation in the kinds of data that actually get produced.

Fang- commented on content/docs/arvo/concepts/scry.md:
```diff
@@ -0,0 +1,77 @@
+
+One further note on `care`s (which can sometimes be confusing): Strictly speaking `care`s specify submodules in the Clay vane. They're used both in Clay scries as well as Clay's `task`s and `gift`s. For example, a `%x` `care` reads the data of a file, a `%p` `care` reads file permissions, and so forth. To see all of Clay's `care`s and what they do, you can refer to Clay's [Scry Reference](/docs/a ...
+
+Some other vanes also make use of `care`s in their scry endpoints. While such vanes don't have corresponding submodules with strictly defined behaviour like Clay, the `care`s still confer the general nature of the endpoint. The most widely used `care` is `%x`, which implies reading data in a general sense. Gall has special handling of `%x` scries as described in the [Gall agents](#gall-agents) se ...
```
To be clear: cares are considered part of the namespace as a whole, not just Clay's. It's just that Clay implements them most thoroughly, and they might've been over-tuned to be good fits for Clay specifically.

Fang- commented on content/docs/arvo/concepts/scry.md:
```diff
@@ -0,0 +1,77 @@
+
+Gall has a single scry endpoint of its own to check for the existence of an agent, but otherwise all Gall scries are passed through to one of the agents it manages. The target agent to scry is specified in place of the `desk` as described in the diagram above. Each Gall agent includes a `+on-peek` arm that defines its own scry endpoints. For example, `%graph-store` has a number of scry endpoints  ...
+
+Gall agents can expose scry endpoints with any `care`, but most commonly they'll take a `%x` `care`. Gall handles `%x` scries specially - it expects an extra field at the end of the `path` that specifies a `mark`. Gall will attempt to perform a `mark` conversion from the `mark` returned by the scry endpoint to the `mark` specified.
```
May or may not be worth reiterating here that this mark is stripped from the path before being sent into `+on-peek`.

Fang- commented on content/docs/arvo/concepts/scry.md:
````diff
@@ -0,0 +1,77 @@
+The case in the beginning says it takes a `%x` `care` and has a `path` of `/keys`. Additionally, it returns data with a `mark` of `%graph-update-2`. Assuming we wanted the `%graph-update-2` `mark` converted to a `%noun` `mark`, our scry would be composed along the lines of:
+
+```hoon
+.^(update:graph-store %gx /(scot %p our)/graph-store/(scot %da now)/keys/noun)
````
This example is kind of silly in that the type given here is the type of the original mark. Might make the mark conversion more obvious by doing `.^(* ...` instead. (But even then, expecting a noun is always valid regardless of the mark.)

Fang- reviewed (approved):
Some comments and nits, but overall looks good! (:

tinnus-napbus committed:
scries: change example mark from noun to json

tinnus-napbus committed:
scries: fix type in example

tinnus-napbus committed:
scries: change diagram; clarify cares

tinnus-napbus committed:
scries: clarify gall mark behaviour

tinnus-napbus committed:
scries: improve wording re: clay and cares

tinnus-napbus commented:
@Fang- I've addressed your comments. Wrt clay and cares, I've reworded it so it's clear they're not a specifically Clay thing, but I've left most of that explanation cos that was something that was quite confusing when I was first learning this stuff

Fang- commented:
Yeah this seems good, thank you!
